doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
```

Comment:
The API should be usable by programs, thus you should somehow specify how a program can get all available API methods (e.g. to check if what it needs is already available). If you return some unspecified plain text or html, its not really useful as API.